tests/unit/server/api/v1/test_datasets.py

```diff
@@ -5878,6 +5880,181 @@ async def test_search_dataset_records_with_response_status_filter(
+                "vector": {"name": vector_settings.name, "value": selected_vector.value},
+            }
+        }
+        response = await async_client.post(
+            f"/api/v1/me/datasets/{dataset.id}/records/search",
+            headers=owner_auth_header,
+            json=query_json,
+            params={"offset": 0, "limit": 10},
+        )
+
+        assert response.status_code == 200
+        response_json = response.json()
+        assert len(response_json["items"]) == 2
+        assert response_json["total"] == 2
+
+        mock_search_engine.search.assert_not_called()
+        mock_search_engine.similarity_search.assert_called_once_with(
+            dataset=dataset,
+            vector_settings=vector_settings,
+            record=None,
+            value=selected_vector.value,
+            query=TextQuery(q="Test query"),
+            order="most_similar",
+            max_results=10,
+            metadata_filters=[],
+            user_response_status_filter=None,
+        )
+
+    async def test_search_dataset_records_with_wrong_vector(
+        self, async_client: "AsyncClient", mock_search_engine: SearchEngine, owner: User, owner_auth_header: dict
+    ):
+        workspace = await WorkspaceFactory.create()
+        dataset, _, records, *_ = await self.create_dataset_with_user_responses(owner, workspace)
+        vector_settings = await VectorSettingsFactory.create(dataset=dataset)
+
+        query_json = {"query": {"vector": {"name": "wrong_vector", "record_id": str(records[0].id)}}}
+
+        response = await async_client.post(
+            f"/api/v1/me/datasets/{dataset.id}/records/search",
+            headers=owner_auth_header,
+            json=query_json,
+            params={"offset": 0, "limit": 10},
+        )
+
+        assert response.status_code == 422
+        response_json = response.json()
+        assert response_json == {"detail": f"Vector `wrong_vector` not found in dataset `{dataset.id}`."}
+
+    async def test_search_dataset_records_with_wrong_vector_record_id(
+        self, async_client: "AsyncClient", mock_search_engine: SearchEngine, owner: User, owner_auth_header: dict
+    ):
+        workspace = await WorkspaceFactory.create()
+        dataset, _, records, *_ = await self.create_dataset_with_user_responses(owner, workspace)
+        vector_settings = await VectorSettingsFactory.create(dataset=dataset)
+        wrong_record_id = str(uuid.uuid4())
+
+        query_json = {"query": {"vector": {"name": vector_settings.name, "record_id": wrong_record_id}}}
+
+        response = await async_client.post(
+            f"/api/v1/me/datasets/{dataset.id}/records/search",
+            headers=owner_auth_header,
+            json=query_json,
+            params={"offset": 0, "limit": 10},
+        )
+
+        assert response.status_code == 422
+        response_json = response.json()
+        assert response_json == {"detail": f"Record with id `{wrong_record_id}` not found in dataset `{dataset.id}`."}
+
```

Comment:
We should create here a `test_search_dataset_records_with_record_id_from_different_dataset` or something similar.